File: just-algos-and-ds/algos/go/algorithms/mcd_math/is_prime.go
package mcd_math

import "math"

func isPrime(n int) bool {
	if n == 1 {
		return false
	}
	if n == 2 {
		return true
	}
	for i := 2; i <= int(math.Floor(float64(n)/2)); i++ {
		if n%i == 0 {
			return false
		}
	}
	return n > 1
}
func isPrimeTwo(n int) bool {
	if n == 1 {
		return false
	}
	if n == 2 {
		return true
	}
	for i := 2; i <= int(math.Floor(math.Sqrt(float64(n)))); i++ {
		if n%i == 0 {
			return false
		}
	}
	return n > 1
}

File: algos/algos-1.0.0/ts/algorithms/math/is_prime/is_prime.ts
export function isPrime(n: number): boolean {
  if (n < 2) {
    return false;
  }
  for (let i = 2; i <= Math.sqrt(n); i++) {
    if (n % i === 0) return false;
  }
  return n > 1;
}

export function isPrimeFunctional(n: number): boolean {
  if (n < 2) {
    return false;
  }
  return !Array.from({length: n - 2}, (_, i) => i + 2).some((i) => n % i === 0);
}

export function isPrimeRecursive(n: number, i = 2): boolean {
  if (n < 2) {
    return false;
  }
  if (i > Math.sqrt(n)) {
    return true;
  }
  if (n % i === 0) {
    return false;
  }
  return isPrimeRecursive(n, i + 1);
}
